wpilib/src/can.rs

```diff
@@ -0,0 +1,201 @@
+use wpilib_sys::*;
+
+pub const TEAM_MANUFACTURER: HAL_CANManufacturer::Type = HAL_CANManufacturer::HAL_CAN_Man_kTeamUse;
+pub const TEAM_DEVICE_TYPE: HAL_CANDeviceType::Type = HAL_CANDeviceType::HAL_CAN_Dev_kMiscellaneous;
+
+#[derive(Debug, Clone)]
+pub struct CanData {
+    data: [u8; 8],
+    length: i32,
+    timestamp: u64,
+}
+
+impl CanData {
+    pub fn data(&self) -> &[u8] {
+        &self.data[..self.length as usize]
+    }
+
+    pub fn raw_data(&self) -> &[u8; 8] {
+        &self.data
+    }
+
+    pub fn into_raw_data(self) -> [u8; 8] {
+        self.data
+    }
+
+    pub fn length(&self) -> i32 {
+        self.length
+    }
+
+    pub fn timestamp(&self) -> u64 {
+        self.timestamp
+    }
+}
+
+/// A device on the CAN bus that conforms to the standard CAN spec
+/// and uses the FRC CAN bus addressing.
+#[derive(Debug)]
+pub struct Can {
+    handle: HAL_CANHandle,
+}
+
+impl Can {
+    pub const TEAM_MANUFACTURER: HAL_CANManufacturer::Type = TEAM_MANUFACTURER;
+    pub const TEAM_DEVICE_TYPE: HAL_CANDeviceType::Type = TEAM_DEVICE_TYPE;
+
+    pub fn new(
+        device_id: i32,
+        device_manufacturer: HAL_CANManufacturer::Type,
+        device_type: HAL_CANDeviceType::Type,
+    ) -> HalResult<Self> {
```

Comment:
Per API guidelines (https://rust-lang-nursery.github.io/api-guidelines/naming.html), I think `new` is supposed to have fewer arguments. However, if `TEAM_MANUFACTURER` refers to a CAN device made _by a team_, then I guess that is the far less common case, and there can't really be a good default for `new`. Not sure what to do about that.